presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2736,7 +2738,7 @@ else if (node.getRowCount() instanceof LongLiteral) {
             return false;
         }
 
-        private OptionalLong analyzeParameterAsRowCount(Parameter parameter, Scope scope)
+        private OptionalLong analyzeParameterAsRowCount(Parameter parameter, Scope scope, String nonConstantErrorContext)
```

Comment:
What does "non constant" refer to in `nonConstantErrorContext`?

Reply:
It refers to a failure to resolve expression as a constant of a given type. I'll rename it to `errorContext`.